memory/table.go

```diff
@@ -1570,11 +1570,13 @@ func (t *Table) CreateIndex(ctx *sql.Context, idx sql.IndexDef) error {
 // DropIndex implements sql.IndexAlterableTable
 func (t *Table) DropIndex(ctx *sql.Context, indexName string) error {
 	for name := range t.indexes {
-		if name == indexName {
+		if strings.ToLower(name) == strings.ToLower(indexName) {
 			delete(t.indexes, name)
+			return nil
 		}
 	}
-	return nil
+	
+	return sql.ErrIndexNotFound.New(indexName)
```

Comment:
test for this one?

Reply:
We have tests of drop index not found, but they happen higher in the stack, in the engine code.

This is just a failsafe